### PR TITLE
fix(unified-search): Refactor header menu code

### PR DESCRIPTION
**Pull request by susnux**
## Summary
1. Adjust styles for the header menu trigger to match the other entries
2. Rename the header menu trigger class so it aligns with other entries
3. Simplify some code (e.g. use `type` on *NcButton* instead of custom styles).

### Screenshots
before|after
---|---
Screenshot_20240120_035700|Screenshot_20240120_040022
Screenshot_20240120_035707|Screenshot_20240120_040029

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

File: core/src/views/UnifiedSearchModal.vue [hunk 1]
```diff
@@ -542,6 +542,7 @@ export default {
 .unified-search-modal {
 	box-sizing: border-box;
 	height: 100%;
+	min-height: 80vh;
```
Comment (emoral435) [inline core/src/views/UnifiedSearchModal.vue:545]: is this not already handled by the modal we already have? I do not mind leaving this here just for safe bet, but I do not know if this contributes much, unless I am wrong

**Review by emoral435 — APPROVED** [review]

File: core/src/views/UnifiedSearchModal.vue (hunk 1 above)
Comment (susnux) [reply core/src/views/UnifiedSearchModal.vue:545]: No it is only done on the FilePicker, NcDialog does not set any height.

Comment (emoral435) [reply core/src/views/UnifiedSearchModal.vue:545]: Ahhhh, I see. I thought this was configured by NcDialog, but this is a nice change then! :)

File: core/src/views/UnifiedSearch.vue [hunk 2]
```diff
@@ -62,35 +65,32 @@ export default {
...
+				box-shadow: none !important;
+			}
+
+			&:not(:hover,:focus,:focus-visible) {
+				opacity: .85;
+			}
+
+			&-icon {
+				// ensure the icon has the correct color
+				color: var(--color-background-plain-text) !important;
```
Comment (Pytal) [inline core/src/views/UnifiedSearch.vue:91]: Note for reviewers: new variable in https://github.com/nextcloud/server/pull/42977

**Review by Pytal — APPROVED** [review]

**Review by nfebe — APPROVED** [review]

File: core/src/views/UnifiedSearch.vue (hunk 2 above)
Comment (susnux) [reply core/src/views/UnifiedSearch.vue:91]: (This was cherry picked from that PR so it gets in here).

But I reverted for current master (use `primary-text`).

**Pull request merged**